src/components/Homepage/components/RestaurantList/RestaurantList.jsx

```diff
@@ -1,25 +1,31 @@
-import React from "react";
-import { ListWrapper } from "./RestaurantListElements";
-import RestaurantCard from "../RestaurantCard";
+import React, { useEffect, useState } from "react";
 import { Link } from "react-router-dom";
 
-export const RestaurantList = (props) => {
-  const RESTAURANTS_API =
-    "https://redi-final-restaurants.herokuapp.com/restaurants";
+import { ContentWrapper } from "../../../../common";
+
+import { ListWrapper, Loading } from "./RestaurantListElements";
+import RestaurantCard from "../RestaurantCard";
 
-  const [restaurantList, setRestaurantList] = React.useState([]);
-  const [searchValue, setSearchValue] = React.useState(props.searchValue);
+const RESTAURANTS_API =
+  "https://redi-final-restaurants.herokuapp.com/restaurants";
 
-  React.useEffect(() => {
-    async function fetchData() {
+export const RestaurantList = (props) => {
+  const [restaurantList, setRestaurantList] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [searchValue, setSearchValue] = useState(props.searchValue);
+
+  useEffect(() => {
+    const fetchData = async () => {
       const response = await fetch(RESTAURANTS_API);
       const data = await response.json();
       setRestaurantList(data.results);
-    }
+    };
+
     fetchData();
+    setLoading(false);
```

Comment:
This function will run before fetching is complete so it will cancel loading state while it's actually loading. I think it's better to move it into `fetchData` function too

Reply:
Moved it, thank you!